config/query_version.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python3
+import argparse, os, subprocess
+
+def _call(command, **kwargs):
+    rslt = subprocess.check_output(command, shell = True, **kwargs)
+    return rslt.decode().rstrip()  # return as str & remove any trailing '\n'
+
+def query_version():
+    version_file = os.path.join(os.path.dirname(__file__), '../VERSION')
+    return _call("tail -1 " + version_file)
```

Comment:
One concern I have is that `subprocess.check_output` will raise an exception if the command results in an error. I'm not sure how universal `tail` is, for example, on windows. I don't know anyone compiling there, but it's possible. It might just be worth putting a try/except in `_call` to return None or something so the build doesn't fail because of this.

Reply:
Good point: `tail` is a standard posix utility, but I have no idea whether it's on windows.

Since the `VERSION` file is a really tiny file, I just replaced the logic with an 8-line python file that basically does the same thing (it tries to be a little more careful in case trailing blank-lines accidentally are introduced to the file)